Client/Frontend/Browser/Tabs/RemoteTabs/RemoteTabsPanel.swift

```diff
@@ -52,8 +52,20 @@ class RemoteTabsPanel: UIViewController,
         }
     }
 
+    // MARK: - Actions
+
+    func tableViewControllerDidPullToRefresh() {
+        refreshTabs()
+    }
+
     // MARK: - Internal Utilities
 
+    private func refreshTabs() {
+        // Ensure we do not already have a refresh in progress
+        guard state.refreshState != .refreshing else { return }
```

Comment:
I think this check should be handled in the reducer.

Reply:
@OrlaM Offhand I think we still need it here anyway for the case in which someone tries to pull-to-refresh while a background sync is already in flight? (I could be mistaken though). I'm happy to move it elsewhere however, LMK if you have a preference on where that should live.

Reply:
Yeah, I just realized the check will need to live in both the reducer and the middleware which is more complex. My concern is that this is essentially making state decisions (deciding to mutate the state or not based on current state) which should be contained within the reducers and middleware. I'm now uncertain and I might be over thinking it. Let me think about it a bit and I'll get back to you. Feel free to merge for now, it can be updated in a future PR if needed.

Reply:
> My concern is that this is essentially making state decisions (deciding to mutate the state or not based on current state) which should be contained within the reducers and middleware. 

Makes sense, I agree that it doesn't feel right for the table VC to be making logical decisions based on the state.

Reply:
(Based on team chat sounds like we'll probably keep this as-is for now but can revisit later if needed)